dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1404,12 +1404,20 @@ public String getWorkflowDag(@ApiParam(hidden = true) @Auth Optional<User> user,
         WorkflowVersion workflowVersion = getWorkflowVersion(workflow, workflowVersionId);
         SourceFile mainDescriptor = getMainDescriptorFile(workflowVersion);
 
+        if (workflowVersion.getCommitID() != null && workflowVersion.getCommitID().equals(workflowVersion.getCommitForJsonGeneration()) && workflowVersion.getDagJson() != null) {
```

Comment:
What is the commit id for a hosted workflow (I don't know)? If it's null, then caching will never work for hosted workflows.